go/plugins/compat_oai/openai/openai_live_test.go

```diff
@@ -215,4 +211,62 @@ func TestPlugin(t *testing.T) {
 
 		t.Logf("system message response: %+v", out)
 	})
+
+	t.Run("generation config", func(t *testing.T) {
+		// Create a config with specific parameters
+		config := &ai.GenerationCommonConfig{
+			Temperature:     0.2,
+			MaxOutputTokens: 50,
+			TopP:            0.5,
+			StopSequences:   []string{".", "!", "?"},
+		}
+
+		resp, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("Write a short sentence about artificial intelligence."),
+			ai.WithConfig(config),
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+		out := resp.Message.Content[0].Text
+		t.Logf("generation config response: %+v", out)
+	})
+
+	t.Run("unsupported config field", func(t *testing.T) {
+		// Create a config with an unsupported TopK parameter
+		config := &ai.GenerationCommonConfig{
+			Temperature:     0.2,
+			MaxOutputTokens: 50,
+			TopK:            10, // TopK is not supported in OpenAI's chat completion API
+		}
+
+		_, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("Write a short sentence about artificial intelligence."),
+			ai.WithConfig(config),
+		)
+		if err == nil {
+			t.Fatal("expected error for unsupported TopK parameter")
+		}
+		if !strings.Contains(err.Error(), "TopK is not supported in OpenAI's chat completion API") {
+			t.Errorf("got error %q, want error containing 'TopK is not supported in OpenAI's chat completion API'", err.Error())
+		}
+		t.Logf("unsupported config error: %v", err)
+	})
+
+	t.Run("invalid config type", func(t *testing.T) {
+		// Try to use a string as config instead of *ai.GenerationCommonConfig
+		config := "not a config"
+
+		_, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("Write a short sentence about artificial intelligence."),
+			ai.WithConfig(config),
+		)
+		if err == nil {
+			t.Fatal("expected error for invalid config type")
+		}
+		if !strings.Contains(err.Error(), "config must be of type *ai.GenerationCommonConfig") {
+			t.Errorf("got error %q, want error containing 'config must be of type *ai.GenerationCommonConfig'", err.Error())
+		}
+		t.Logf("invalid config type error: %v", err)
+	})
```

Comment:
Illustrates user behavior with a config of an improper type.

Reply:
The UI may send some fields that not be supported. Take a look with a googlegenai.

Reply:
Working with map conversion from ui